library/src/main/scala/scala/scalajs/js/JSConverters.scala

```diff
@@ -51,4 +62,35 @@ object JSConverters {
       arr: scala.Array[T]): JSRichGenTraversableOnce[T] =
     new JSRichGenTraversableOnce(arr)
 
+  @inline
+  implicit def JSRichFutureThenable[A](f: Future[Thenable[A]]): JSRichFuture[A] =
+    new JSRichFuture[A](f.asInstanceOf[Future[A | Thenable[A]]])
```

Comment:
A good read about these problems is the es-discuss thread: https://esdiscuss.org/topic/promise-cast-and-promise-resolve
In particular, pay attention to Andreas Rossberg's interventions, e.g., at https://esdiscuss.org/topic/promise-cast-and-promise-resolve#content-12